src/FunctionLoader.cs

```diff
@@ -47,54 +105,156 @@ internal class AzFunctionInfo
         internal const string TriggerMetadata = "TriggerMetadata";
         internal const string DollarReturn = "$return";
 
-        internal readonly string Directory;
+        internal readonly string FuncDirectory;
+        internal readonly string FuncName;
         internal readonly string EntryPoint;
-        internal readonly string FunctionName;
         internal readonly string ScriptPath;
+        internal readonly HashSet<string> FuncParameters;
         internal readonly AzFunctionType Type;
-        internal readonly MapField<string, BindingInfo> AllBindings;
-        internal readonly MapField<string, BindingInfo> OutputBindings;
+        internal readonly ReadOnlyDictionary<string, ReadOnlyBindingInfo> AllBindings;
+        internal readonly ReadOnlyDictionary<string, ReadOnlyBindingInfo> InputBindings;
+        internal readonly ReadOnlyDictionary<string, ReadOnlyBindingInfo> OutputBindings;
 
+        /// <summary>
+        /// Construct an object of AzFunctionInfo from the 'RpcFunctionMetadata'.
+        /// Necessary validations are done on the metadata and script.
+        /// </summary>
         internal AzFunctionInfo(RpcFunctionMetadata metadata)
```

Comment:
This class is getting big. Can we pull it out?

Reply:
It makes sense to pull it to a separate file. I will deffer that to another PR to make it easier to review changes to this type.